notes.go

```diff
@@ -0,0 +1,211 @@
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.CreateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		return
+	}
+
+	res, err := h.notesClient.CreateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) GetNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.GetNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.GetNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) UpdateNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.UpdateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		c.JSON(http.StatusOK, httpError{Error: err.Error()})
+		return
+	}
+	body.Id = c.Param("note_id")
+
+	res, err := h.notesClient.UpdateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) DeleteNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.DeleteNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.DeleteNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ListNotes(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.ListNotesRequest{
+		AuthorId: c.Query("author_id"),
+	}
+
+	res, err := h.notesClient.ListNotes(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ExportNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	formatMap := map[string]notesv1.NoteExportFormat{
+		"":    notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_INVALID,
+		"md":  notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_MARKDOWN,
+		"pdf": notesv1.NoteExportFormat_NOTE_EXPORT_FORMAT_PDF,
+	}
+
+	body := &notesv1.ExportNoteRequest{
+		ExportFormat: notesv1.NoteExportFormat(formatMap[c.Query("format")]),
```

Comment:
I think we should check if the map value exists
```go
val, ok := formatMap[c.Query("format")]
if ok {
    // Value exists
}
```
Idk how it would react if we put some shit as a query

Idk tell my what u think about it